tempsensord/sensor_daemon.cpp

```diff
@@ -139,13 +148,25 @@ int main(void) {
                 humid_buf[6] = '\0';
                 temp_buf[6] = '\0';
 
+                //Update json message
+                strcpy(msg, "{\"temperature\":");
+                strcat(msg, temp_buf);
+                strcat(msg, ",\"humidity\":");
+                strcat(msg, humid_buf);
+                strcat(msg, "}\0");
+                msg_len = strlen(msg);
+
                 //Print buffer values (Will be removed when )
-                printf("Humid Buffer: \"%s\" \n", humid_buf);
-                printf("Temp Buffer: \"%s\" \n", temp_buf);
+                //printf("Humid Buffer: \"%s\" \n", humid_buf);
+                //printf("Temp Buffer: \"%s\" \n", temp_buf);
+
+                //Send update over ZMQ to server
+                zmq_send(responder, msg, msg_len, 0);

```

Comment:
as with my other comment, have this end determine the send frequency. probably easiest to set up another thread to send and use a lock of some sort for accessing the buffers. or just keep copying stuff, who cares about memory usage.

Reply:
Instead of making a new thread for this I can just adjust the frequency that the Arduino sends the data back to the Raspberry Pi. This will effectively do the same thing without adding the extra hassle of making a new thread and setting up a mutex lock.